Src/FluentAssertions/Equivalency/DataRowEquivalencyStep.cs

```diff
@@ -0,0 +1,214 @@
+             || ((dataRowConfig != null) && dataRowConfig.ExcludeOriginalData))
+            {
+                compareOriginalVersions = false;
+            }
+
+            foreach (var columnName in expectationColumnNames.Union(subjectColumnNames))
+            {
+                var expectationColumn = expectation.Table.Columns[columnName];
+                var subjectColumn = subject.Table.Columns[columnName];
+
+                if (((dataSetConfig != null) && dataSetConfig.ShouldExcludeColumn(subjectColumn))
+                 || ((dataTableConfig != null) && dataTableConfig.ShouldExcludeColumn(subjectColumn))
+                 || ((dataRowConfig != null) && dataRowConfig.ShouldExcludeColumn(subjectColumn)))
+                {
+                    continue;
+                }
+
+                if (!ignoreUnmatchedColumns)
+                {
+                    AssertionScope.Current
+                        .ForCondition(subjectColumn != null)
+                        .FailWith("Expected {context:DataRow} to have column '{0}'{reason}, but found none", columnName);
+
+                    AssertionScope.Current
+                        .ForCondition(expectationColumn != null)
+                        .FailWith("Found unexpected column '{0}' in {context:DataRow}", columnName);
+                }
+
+                if ((subjectColumn != null) && (expectationColumn != null))
+                {
+                    CompareFieldValue(context, parent, subject, expectation, subjectColumn, subjectVersion, expectationColumn, expectationVersion);
+
+                    if (compareOriginalVersions)
+                    {
+                        CompareFieldValue(context, parent, subject, expectation, subjectColumn, DataRowVersion.Original, expectationColumn, DataRowVersion.Original);
+                    }
+                }
+            }
+        }
+
+        private static void CompareFieldValue(IEquivalencyValidationContext context, IEquivalencyValidator parent, DataRow subject, DataRow expectation, DataColumn subjectColumn, DataRowVersion subjectVersion, DataColumn expectationColumn, DataRowVersion expectationVersion)
+        {
+            var nestedContext = context.AsCollectionItem(
+                subjectVersion == DataRowVersion.Current
+                ? subjectColumn.ColumnName
+                : $"{subjectColumn.ColumnName}, DataRowVersion.Original",
+                subject[subjectColumn, subjectVersion],
+                expectation[expectationColumn, expectationVersion]);
+
+            if (nestedContext != null)
+            {
+                parent.AssertEqualityUsing(nestedContext);
+            }
+        }
+
+        private class SelectedDataRowMembers
+        {
+            [SuppressMessage("StyleCop.CSharp.MaintainabilityRules", "SA1401:FieldsMustBePrivate", Justification = "Field exists for performance")]
+            internal bool HasErrors;
```

Comment:
How is this more performant than a regular property?

Reply:
Maybe it isn't. I have habits from days of programming in much more explicit languages against much more constrained systems, and without thinking about it want to avoid using a property because of it being a function call under the hood. But, for a private class, that's likely to be optimized away, and in any case even if it does make a function call, that's not likely to be measurable overhead.